Makefile

```diff
@@ -5,9 +5,16 @@ default: fmtcheck vet build
 
 # test runs the test suite and vets the code
 test: fmtcheck
-	@echo "==> Running Tests"
+	@echo "==> Running Unit Tests"
+	cd govcd && go test -v -tags "unit" .
+	@echo "==> Running Functional Tests"
```

Comment:
I don't remember what the default behavior for Makefile is. 
What happens if the unit tests fail? Does it continue or does it fail?
Should we consider a `scripts/runtest.sh` file to manage complexity like we have in terraform-provider-vcd?

Reply:
I could probably add that if needed.

It exits. I tested (inserted `t.Error()` in the test) it:
```
$  make test
==> Checking that code complies with gofmt requirements...
==> Running Unit Tests
cd govcd && go test -v -tags "unit" .
=== RUN   TestTags
--- PASS: TestTags (0.00s)
=== RUN   Test_updateNicParameters_multiNIC
=== RUN   Test_updateNicParameters_multiNIC/Ordered_NICs_list
=== RUN   Test_updateNicParameters_multiNIC/Unordered_NIC_list
--- FAIL: Test_updateNicParameters_multiNIC (0.00s)
    vm_unit_test.go:99:
    --- PASS: Test_updateNicParameters_multiNIC/Ordered_NICs_list (0.00s)
    --- PASS: Test_updateNicParameters_multiNIC/Unordered_NIC_list (0.00s)
=== RUN   Test_updateNicParameters_singleNIC
=== RUN   Test_updateNicParameters_singleNIC/IPAllocationModeDHCP
=== RUN   Test_updateNicParameters_singleNIC/IPAllocationModePool
=== RUN   Test_updateNicParameters_singleNIC/IPAllocationModeNone
=== RUN   Test_updateNicParameters_singleNIC/IPAllocationModeManual
=== RUN   Test_updateNicParameters_singleNIC/IPAllocationModeDHCPInvalidIP
--- PASS: Test_updateNicParameters_singleNIC (0.00s)
    --- PASS: Test_updateNicParameters_singleNIC/IPAllocationModeDHCP (0.00s)
    --- PASS: Test_updateNicParameters_singleNIC/IPAllocationModePool (0.00s)
    --- PASS: Test_updateNicParameters_singleNIC/IPAllocationModeNone (0.00s)
    --- PASS: Test_updateNicParameters_singleNIC/IPAllocationModeManual (0.00s)
    --- PASS: Test_updateNicParameters_singleNIC/IPAllocationModeDHCPInvalidIP (0.00s)
FAIL
FAIL    github.com/vmware/go-vcloud-director/v2/govcd   0.014s
make: *** [test] Error 1
```